maven-tools/fragmentmerge-maven-plugin/src/main/java/org/jaffa/plugins/FragmentMergeMojo.java

```diff
@@ -0,0 +1,159 @@
+
+
+/**
+ * Maven Plugin to merge the resource files during Maven Life Cycle Phase of Process Classes and place them under META-INF
+ */
+@Mojo(name="fragmentmerge", defaultPhase = LifecyclePhase.PROCESS_CLASSES, requiresDependencyResolution = ResolutionScope.RUNTIME)
+public class FragmentMergeMojo extends AbstractMojo{
+
+
+    private static final String META_INF_LOCATION = File.separator+"META-INF"+File.separator;
+
+    private static final String PROPERTIES_FILE = "resources.properties";
+
+    private static final String DWR_FILE = "dwr.xml";
+
+    private static final String JAWR_FILE = "jawr.properties";
+
+
+    /**
+     * Source for type of resources to look for
+     */
+    @Parameter
+    File resources;
+
+
+    /**
+     * Directory containing the generated JAR.
+     */
+    @Parameter( defaultValue = "${project.build.directory}", required = true )
+    private File outputDirectory;
+
+    /**
+     * Directory containing the classes and resource files that should be packaged into the JAR.
+     */
+    @Parameter( defaultValue = "${project.build.outputDirectory}", required = true )
+    private File classesDirectory;
+
+    /**
+     * The {@link {MavenProject}.
+     */
+    @Parameter( defaultValue = "${project}", readonly = true, required = true )
+    private MavenProject project;
+
+    /**
+     * The {@link MavenSession}.
+     */
+    @Parameter( defaultValue = "${session}", readonly = true, required = true )
+    private MavenSession session;
+
+
+    public void execute() throws MojoExecutionException {
+        getLog().info("Initialize Fragment Merging Process");
+        try {
+
+            File target = new File(project.getBuild().getDirectory());
+
+            File resources = new File(classesDirectory + META_INF_LOCATION + PROPERTIES_FILE);
+            File dwr = new File(classesDirectory + META_INF_LOCATION + DWR_FILE);
+            File jawr = new File(classesDirectory + META_INF_LOCATION + JAWR_FILE);
+
+            if (target.exists()) {
+                Collection<File> fragFiles = FileUtils.listFiles(new File(project.getBuild().getDirectory()), new String[]{"pfragment", "xfragment"}, true);
+
+                getLog().info("Files in build directory " + target.getAbsolutePath() + " : " + fragFiles.size());
+                Iterator<File> iter = fragFiles.iterator();
+                while (iter.hasNext()) {
+                    File frag = iter.next();
+                    if (frag.getName() != null && frag.getName().startsWith("ApplicationResources")) {
+                        AppResourceFragments.merge(resources, frag);
+                    } else if (frag.getName() != null && frag.getName().startsWith("dwr")) {
+                        DwrFragments.merge(dwr, frag);
+                    } else if (frag.getName() != null && frag.getName().startsWith("jawr")) {
+                        JawrResourceFragments.merge(jawr, frag);
+                    }
+                    frag.delete();
```

Comment:
No, this is deleting file from 'target'. This plugin will execute after the classes are compiled and before the jar is built. We wanted to delete the files from their original location and merge them under META-INF when they go inside jar